test-auth.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+set -o errexit
+set -o nounset
+
+IMG="$1"
+TAG="$2"
+
+case "$TAG" in
+  4.1 )
+    ES_VERSION=1.5
+    ;;
+  4.4 )
+    ES_VERSION=2.2
+    ;;
+  * )
+    ES_VERSION="$TAG"
+    ;;
+esac
+
+DB_CONTAINER="kbelastic"
+DATA_CONTAINER="${DB_CONTAINER}-data"
+ES_IMG="quay.io/aptible/elasticsearch:${ES_VERSION}"
+KIBANA_CONTAINER="kibana-${TAG}"
+KIBANA_CREDS=kuser:kpass
+ES_USER=euser 
+ES_PASS=epass
+
+function cleanup {
+  echo "Cleaning up"
+  docker rm -f "$DB_CONTAINER" "$DATA_CONTAINER" "$KIBANA_CONTAINER" > /dev/null 2>&1 || true
+}
+
+function wait_for_request {
+  CONTAINER=$1
+  REQUEST=$2
+  for _ in $(seq 1 30); do
+    if docker exec -it "$CONTAINER" curl -fs "$REQUEST" > /dev/null 2>&1; then
+      return 0
+    fi
+    sleep 1
+  done
+
+  echo "No response"
+  docker logs "$CONTAINER"
+  return 1
+}
+
+trap cleanup EXIT
+cleanup
+
+echo "Creating data container"
+docker create --name "$DATA_CONTAINER" "${ES_IMG}"
+
+echo "Starting DB"
+docker run -it --rm \
+  -e USERNAME=${ES_USER} -e PASSPHRASE=${ES_PASS} -e DATABASE=db \
+  --volumes-from "$DATA_CONTAINER" \
+  "${ES_IMG}" --initialize \
+  >/dev/null 2>&1
+
+docker run -d --name="$DB_CONTAINER" \
+  --volumes-from "$DATA_CONTAINER" \
+  "${ES_IMG}"
+
+echo "Waiting for DB to come online"
+wait_for_request "$DB_CONTAINER" "http://localhost:9200"
+
+ES_IP=$(docker inspect ${DB_CONTAINER} | grep '\"IPAddress\"' | head -n 1 | awk -F "\"" '{print $4}')
+
+echo "Verifying DB shutdown message isn't present"
+docker logs "$DB_CONTAINER" 2>&1 | grep -vqiE "node.+closed"
+
+echo "Sarting Kibana"
+docker run -d --name="$KIBANA_CONTAINER" \
+  -e DATABASE_URL="http://${ES_USER}:${ES_PASS}@${ES_IP}:80" \
+  -e AUTH_CREDENTIALS="$KIBANA_CREDS" \
+  -e TESTING="true" \
+  "$IMG" 
+
+wait_for_request "${KIBANA_CONTAINER}" "http://localhost:5601/elasticsearch/"
```

Comment:
Does this actually test anything? On 6.1  at the very least, it passes even if I comment out the auth headers required to talk to ES (this URL simply returns a redirect).

I looked into the requests made by Kibana to ES. Here is a minimal test that actually does something useful. This appears to work on >= 4.4 (can you try and add another one that works for 4.1?):

```
wait_for_request "${KIBANA_CONTAINER}" \
  'http://localhost:5601/elasticsearch/*/_search' \
  -H "kbn-version: ${KIBANA_VERSION}" \
  -H 'content-type: application/json' \
  --data-binary '{}'
```

To make this work, you need `wait_for_request` to accept `$@`, like this:

```
function wait_for_request {
  CONTAINER=$1
  for _ in $(seq 1 30); do
    if docker exec -it "$CONTAINER" curl -f -v "$@"; then
      return 0
    fi
    sleep 1
  done

  echo "No response"
  docker logs "$CONTAINER"
  return 1
}
```

You also  need to pass `KIBANA_VERSION` to this test.